scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1516,6 +1536,152 @@
                    max-considerable)
                  next-considerable))))))
 
+(defn job->task-txn
+  "Generate task transaction for a job."
+  [job metadata compute-cluster]
+  (let [{:keys [job/last-waiting-start-time job/uuid]} job
+        {:keys [executor task-id]} metadata
+        job-ref [:job/uuid uuid]
+        instance-start-time (now)]
+    [[:job/allowed-to-start? job-ref]
+     ; The job will remain waiting until the instance is running.
+     [:db/add job-ref :job/state :job.state/waiting]
+     (cond->
+      {:db/id (d/tempid :db.part/user)
+       :job/_instance job-ref
+       :instance/executor executor
+       :instance/executor-id task-id ; NB command executor uses the task-id as the executor-id 
+       :instance/hostname "Unknown"
+       :instance/ports 9876
+       :instance/preempted? false
+       :instance/progress 0
+       :instance/slave-id "Unknown"
```

Comment:
Why not leave this empty and unspecified?

Reply:
Done